### PR TITLE
Support changing namecolors through the loginserver

### PR DESCRIPTION
**Pull request by mia-pi-git**
Approved by US.
I specifically routed this through the loginserver in order to have it write to the usermodlog (requested by US).
This reads/writes to a `config/colors.json` file that clients request on startup, and then assign to the existing colors (I specifically chose a new JSON file because editing the config directly programatically would be a nightmare).
I'll be setting up the server-side command code shortly.

### CODE REVIEW

File: lib/dispatcher.lib.php [hunk 1]
```diff
@@ -112,7 +112,7 @@ public function findServer() {
 			}
 			return null;
 		} else {
-			$server =& $PokemonServers[$serverid];
+			$server = &$PokemonServers[$serverid];
```
Comment (AnnikaCodes) [inline lib/dispatcher.lib.php:115]: is this change necessary?

File: lib/dispatcher.lib.php [hunk 2]
```diff
@@ -356,6 +356,65 @@ public function updateuserstats($dispatcher, &$reqData, &$out) {
 		$dispatcher->setPrefix(''); // No need for prefix since only usable by server.
 	}
 
+	public function updatenamecolor($dispatcher, &$reqData, &$out) {
```
Comment (AnnikaCodes) [inline lib/dispatcher.lib.php:359]: where is the permissions check for this?

Comment (mia-pi-git) [reply lib/dispatcher.lib.php:359]: The permissions check on this end validates that it comes from sim3. Sim3 itself can handle permissions checking, since only that server (including an IP check) can send said request.

File: lib/dispatcher.lib.php (hunk 1 above)
Comment (mia-pi-git) [reply lib/dispatcher.lib.php:115]: Nope. Entirely just a nitpick I saw while I was digging around.

File: lib/dispatcher.lib.php (hunk 2 above)
Comment (mia-pi-git) [reply lib/dispatcher.lib.php:359]: Yeah, it validates both the server token and the IP address the request is coming from.

File: lib/dispatcher.lib.php (hunk 1 above)
Comment (mia-pi-git) [inline lib/dispatcher.lib.php:115]:
```suggestion
			$server = & $PokemonServers[$serverid];
```
